pages/api/retrieval.ts

```diff
@@ -0,0 +1,169 @@
+      IndexName: "retrieved-index",
+      Limit: maxRecords,
+      //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
+      KeyConditionExpression: "Retrieved = :zero",
+      ExpressionAttributeValues: { ":formID": { S: formID }, ":zero": { N: "0" } },
+      //A filter expression cannot contain partition key or sort key attributes.
+      //You need to specify those attributes in the key condition expression, not the filter expression.
+      FilterExpression: "FormID = :formID",
+      ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved",
+    };
+    //Get form's responses for formID
+    const formResponses = await db.send(new QueryCommand(getItemsDbParams));
+    //Collecting items submissionIDs for logging and updating items.
+    submissionIDlist =
+      formResponses?.Items?.map((response) => {
+        return response.SubmissionID.S;
+      }) ?? [];
+    if (formResponses.Count && formResponses.Count > 0) {
+      //Preparing a query to set records's attribute Retrieved to 1.
+      for (const submissionID of submissionIDlist as string[]) {
+        //Create a new update object statement
+        const updateItem = {
+          TableName: "Vault",
+          Key: {
+            SubmissionID: {
+              S: submissionID,
+            },
+            FormID: {
+              S: formID,
+            },
+          },
+          UpdateExpression: "SET Retrieved = :retrieved",
+          ConditionExpression: "SubmissionID = :submission",
+          ExpressionAttributeValues: {
+            ":retrieved": { N: "1" },
+            ":submission": { S: submissionID },
+          },
+          ReturnValues: "NONE",
+        };
+        //Update one item at the time
+        await db.send(new UpdateItemCommand(updateItem));
+      }
+    }
+    //Return responses data
+    return res.status(200).json({ responses: formResponses.Items });
+  } catch (error) {
+    logMessage.warn(`Possible data loss for these submissionIDs`);
+    //print submission ids as string
+    logMessage.warn(submissionIDlist ? submissionIDlist.toString() : "Empty submissionID list");
+    logMessage.error(error);
+    return res.status(500).json({ responses: "Error on Server Side" });
+  }
+}
+
+/**
+ * @description
+ * This function will make sure that the incoming request is valid by enforcing a set of
+ * rules.
+ * Request parameters :
+ *  - The maxRecords exists and it ranges between 1 and 10.
+ *  - A formID exists and is valid i.e 78
+ * BearerToken :
+ *  - Must contains an email address
+ *  - It has not expired
+ *  - It exists in our Database
+ *  - The record found in the database includes the email, active, and template_id fields
+ *  - The email on the token record matches the email in the payload.
+ *  - The active flag boolean is true.
+ * @param handler - A method to retrieve form's responses
+ * @returns
+ */
+export const formResponsesReqValidator = (
```

Comment:
This should have been in a separate file as with mostly all middleware functions for reusability

Reply:
Done! It was renamed as  `checkIfValidTemporaryToken`